backend/kesaseteli/applications/tests/test_youth_applications_api.py

```diff
@@ -1908,55 +1938,67 @@ def get_expected_reason(
     "same_social_security_number,"
     "is_existing_active,"
     "is_existing_expired,"
-    "is_existing_rejected,"
+    "is_existing_current_year,"
+    "existing_status,"
     "expected_reason",
     [
         (
             same_email,
             same_social_security_number,
             is_existing_active,
             is_existing_expired,
-            is_existing_rejected,
+            is_existing_current_year,
+            existing_status,
             get_expected_reason(
                 same_email,
                 same_social_security_number,
                 is_existing_active,
                 is_existing_expired,
-                is_existing_rejected,
+                is_existing_current_year,
+                existing_status,
             ),
         )
         for same_email in [False, True]
         for same_social_security_number in [False, True]
         for is_existing_active in [False, True]
         for is_existing_expired in [False, True]
-        for is_existing_rejected in [False, True]
+        for is_existing_current_year in [False, True]
+        for existing_status in [
+            YouthApplicationStatus.ACCEPTED,
+            YouthApplicationStatus.REJECTED,
+        ]
     ],
 )
 def test_youth_application_post_error_codes(
     api_client,
-    same_email,
-    same_social_security_number,
-    is_existing_active,
-    is_existing_expired,
-    is_existing_rejected,
+    settings,
+    same_email: bool,
+    same_social_security_number: bool,
+    is_existing_active: bool,
+    is_existing_expired: bool,
+    is_existing_current_year: bool,
+    existing_status: YouthApplicationStatus,
     expected_reason,
```

Comment:
This test method is quite long. It could be for example divided into two different tests where expected reason is None or something else. But let's not do this now as it might be a lot of work and we have tight time schedule.

Reply:
[Split the test](https://github.com/City-of-Helsinki/yjdh/compare/4371d30673a69936ccfeafbced2d77d3a4e0f3d7..3caec45addd2e149b0c0f98e2fd8567f6f20cf53)